### PR TITLE
Allow custom layout managers for existing layout types in Controls

### LINKED ISSUE #9010: Allow layout manager replacement for existing layout types
All of the MAUI layouts use a separate LayoutManager which handles the actual measure/arrange logic for that layout type. At the moment, that manager is determined by the `CreateLayoutManager` method on each layout type. `CreateLayoutManager` is protected virtual, so custom layout types can specify their own layout manager. 

This works fine for users who want to create custom layout types. However, if a user wants to use an existing layout type and simply customize the behavior of the manager, that user is forced to subclass the existing type. It's not difficult to do, and the code is minimal. But for users who are converting an existing codebase and who want a custom behavior, this can mean updating the type on a large number of layouts. The effort is even greater if they have a lot of XAML layouts to update, since they may also need to add namespace declarations for their custom layout type. 

To address this, I'm proposing that we introduce a static layout manager factory concept similar to what we have in `ViewHandler` - an optional static factory that users can configure in their application which will handle creating layout managers. If the factory is not specified, it will fall through to the current behavior. If the factory is specified, it will be given the opportunity to create a layout manager for the given layout. 

This way, users can modify the layout behavior of existing layout types without having to subclass (much as they can with other controls).

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

Service-based implementation of #9010

### Issues Fixed

Fixes #9010

### CODE REVIEW

File: src/Controls/src/Core/Layout/Layout.cs [hunk 1]
```diff
@@ -14,7 +16,19 @@ public abstract partial class Layout : View, Maui.ILayout, IList<IView>, IBindab
...
+		{
+			get
+			{
+				return _layoutManager ??= GetLayoutManagerFromFactory(this) ?? CreateLayoutManager();
+			}
+		}
+
+		static ILayoutManager GetLayoutManagerFromFactory(Layout layout) 
+		{
+			var factory = layout.Handler?.MauiContext?.Services?.GetService<ILayoutManagerFactory>();
```
Comment (PureWeen) [inline src/Controls/src/Core/Layout/Layout.cs:29]:
```suggestion
			var factory = layout.FindMauiContext()?.Services?.GetService<ILayoutManagerFactory>();
```

Comment (hartez) [reply src/Controls/src/Core/Layout/Layout.cs:29]: TIL

File: src/Controls/src/Core/Layout/ILayoutManagerFactory.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.Maui.Layouts;
+
+namespace Microsoft.Maui.Controls
+{
+	public interface ILayoutManagerFactory 
+	{
+		ILayoutManager CreateLayoutManager(Layout layout);
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Layout/ILayoutManagerFactory.cs:7]: Some public APIs such as tizen remain to be updated.

**Comment by PureWeen** [comment]
At a later point it might be useful to add some docs around this and other layout features

https://github.com/dotnet/maui/tree/main/docs/design

**Review by PureWeen — APPROVED** [review]

**Pull request merged**